js/script.js

```diff
@@ -1,16 +1,255 @@
-// Javascript Code.
-angular.module('app', [])
-.controller('PasswordController', function PasswordController($scope) {
-  $scope.title = "This is AngularJS and Jasmin!";
-  $scope.password = '';
-  $scope.grade = function() {
-    var size = $scope.password.length;
-    if (size > 8) {
-      $scope.strength = 'strong';
-    } else if (size > 3) {
-      $scope.strength = 'medium';
-    } else {
-      $scope.strength = 'weak';
-    }
-  };
+
+//funciones jasmine
+function checkLength(clave){
+	if(clave.length>8){
+		return true;
+	}else{
+		return false;
+	};
+}
+function checkUpper(clave){
+	if(/[A-Z]/.test(clave)){
+		return true;
+	}else{
+		return false;
+	};
+}
+function checkLower(clave){
+	if(/[a-z]/.test(clave)){
+		return true;
+	}else{
+		return false;
+	};
+}
+function checkNumber(clave){
+	if(/\d/.test(clave)){
+		return true;
+	}else{
+		return false;
+	};
+}
+//Starting parse
+Parse.initialize("o738tDIjX7Oq1jSB1PtSG6LfVeZqOgpaKH0pK3dt", "p7JfKdqPlYwWoenFcH1pnxR73YDzNaHAjz6iAwhq");
+//Starting angular and setting routes
+
+(function(){
+var pokemonModule = angular.module('pokemonModule', ["ngRoute"]).config(function($routeProvider) {
+	$routeProvider
+	.when('/', {
+		templateUrl: 'login.html'
+	})
+	.when('/paginaInicio', {
+		templateUrl: 'paginaInicio.html'
+	})
+	.when('/FirstGeneration', {
+		templateUrl: 'FirstGeneration.html'
+	})
+	.when('/SecondGeneration', {
+		templateUrl: 'SecondGeneration.html'
+	})
+	.when('/ThirdGeneration', {
+		templateUrl: 'ThirdGeneration.html'
+	})
+		.when('/FourthGeneration', {
+		templateUrl: 'FourthGeneration.html'
+	})
+		.when('/FifthGeneration', {
+		templateUrl: 'FifthGeneration.html'
+	})
+		.when('/SixthGeneration', {
+		templateUrl: 'SixthGeneration.html'
+	})
+		.when('/FullPokedex', {
+		templateUrl: 'pokedexFull.html'
+	})
+	.otherwise({
+		redirectTo: '/'
+	});
 });
+//controler pokemon list
+	pokemonModule.controller("pokeDex",function($scope,$http){
+		
+		//list the generationes pokemon
+	$scope.listPokemonFirstGeneration=[];
+	$scope.listPokemonSecondGeneration= [];
+	$scope.listPokemonThirdGeneration= [];
+	$scope.listPokemonFourthGeneration = [];
+	$scope.listPokemonFifthGeneration = [];
+	$scope.listPokemonSixthGeneration = [];
+	$scope.listPokemonFullPokedex =  [];
+    /*for para llamar todos los links de cada pokemon*/
+/**************************************************************/
+
+    for(x=1;x<=151;x++)
```

Comment:
Porque hay tantos for loops? Se deberia poder unificar y simplificar el codigo de alguna forma. Posiblemente tener solo 1 for loop y usar variables adentro para hacer la separacion de generaciones de pokemon. Por ejemplo:
```
pokemonsTotal = 781;
primeraGenFinal = 151;
segundaGenFinal = 251;

for (x=1; x<=pokemonsTotal; x++) {
    if( x <= primeraGenFInal) {
    ...
    } else if (x <= segundaGenFinal) {
    ...
    }
}